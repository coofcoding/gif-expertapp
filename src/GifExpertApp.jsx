import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['fnaf']);

    const onAddCategory = (newCategory) => {
        const lowerCategories = categories.map(category => category.toLowerCase());

        if (lowerCategories.includes(newCategory)) return;

        setCategories([newCategory, ...categories]);
    };

    return (
        <>
            <div className="search-bar">
                <h1>Search Gifs</h1>

                <AddCategory
                    onNewCategory={onAddCategory}
                />
            </div>

            {
                categories.map(category =>
                (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }
        </>
    )
}