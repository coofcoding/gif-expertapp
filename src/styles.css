@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700;800;900&display=swap');

:root {
    --bg-color: #000000;
    --bg-focus-input: #1c092887;
    --section: #ffffff11;
    --placeholder-input: #ffffff32;
    --text-input: #ffffffbf;
    --border-input: #ac3ff687;
    --shadow-input: #8e0beb87;
}

body {
    background-color: rgb(0, 0, 0);
}

#root {
    max-width: 100vw;
}

* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Poppins', sans-serif;
}

.search-bar {
    height: 100vh;
    display: grid;
    place-content: center;
    background-color: var(--bg-color);
    color: #fff;

    & h1 {
        font-size: 100px;
        text-shadow:
            0 0 50px var(--shadow-input),
            0 0 100px var(--shadow-input);
    }

    & input {
        border: none;
        width: 100%;
        font-size: 20px;
        padding: 10px 20px;
        border-radius: 5px;
        font-weight: 500;
        background-color: var(--section);
        border: 2px solid var(--section);
        color: var(--text-input);
        transition: all 100ms;

        &::placeholder {
            color: var(--placeholder-input);
        }

        &:focus {
            outline: none;
            border-color: var(--border-input);
            box-shadow: 0 0 10px var(--shadow-input);
            background-color: var(--bg-focus-input);
        }
    }
}

.category-section {
    margin: 20px 250px;
    border-radius: 20px;
    padding: 30px;
    background-color: var(--section);
    border: 2px solid var(--section);

    & h3 {
        color: #ffffff;
        font-size: 50px;
        padding-bottom: 30px;

        & .category-style {
            font-weight: 200;
            color: var(--placeholder-input);
        }
    }

    & .card-grid {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;

        & .card {
            margin: 0 4px 8px 6px;
            background-color: var(--section);
            padding: 15px;
            border-radius: 10px;
            border: 2px solid var(--section);
            cursor: pointer;
            text-decoration: none;
            transition: all 200ms;

            & p {
                color: #fff;
                font-weight: 600;
                font-size: 12px;
                max-width: 20ch;
            }

            & img {
                width: 200px;
                height: 200px;
                object-fit: cover;
                border-radius: 10px;
                filter: grayscale(100%);
                transition: all 200ms;

                &:hover {
                    filter: grayscale(0);
                }
            }

            &:hover {
                transform: scale(1.03);
                border-color: var(--border-input);
                box-shadow: 0 0 10px var(--shadow-input);
                background-color: var(--bg-focus-input);
            }
        }
    }
}

.loading {
    color: #fff;
    font-weight: 300;
    animation: blinking 1000ms alternate infinite ease-in-out;
}

@keyframes blinking {
    from {
        opacity: 50%;
    }

    to {
        opacity: 15%;
    }
}

@media (max-width: 1068px) {
    .category-section {
        margin: 20px 5%;
    }
}

@media (max-width: 630px) {
    .category-section {
        margin: 20px 0;
    }
}

@media (max-width: 630px) {
    h1 {
        text-align: center;
        font-size: 80px !important;
    }
}

@media (max-width: 568px) {

    h1 {
        text-align: center;
        font-size: 50px !important;
    }

    h3 {
        color: #ffffff;
        font-size: 30px !important;
        padding-bottom: 30px;

        & .category-style {
            font-weight: 200;
            color: var(--placeholder-input);
        }
    }

    .card-grid {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;

        & .card {
            margin: 0 4px 8px 6px;
            background-color: var(--section);
            padding: 15px;
            border-radius: 10px;
            border: 2px solid var(--section);
            cursor: pointer;
            text-decoration: none;
            transition: all 200ms;

            & p {
                color: #fff;
                font-weight: 600;
                font-size: 12px;
                max-width: 20ch;
            }

            & img {
                width: 200px;
                height: 200px;
                object-fit: cover;
                border-radius: 10px;
                filter: grayscale(100%);
                transition: all 200ms;

                &:hover {
                    filter: grayscale(0);
                }
            }

            &:hover {
                transform: scale(1.03);
                border-color: var(--border-input);
                box-shadow: 0 0 10px var(--shadow-input);
                background-color: var(--bg-focus-input);
            }
        }
    }
}