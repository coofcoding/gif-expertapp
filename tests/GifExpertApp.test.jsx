import { fireEvent, render, renderHook, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"
import React, { useState } from "react"

// jest.mock('../src/GifExpertApp')

describe('Pruebas en <GifExpertApp/>', () => {
    const inputValue = 'memes';

    test('actualizar valor del input', async () => {

        render(
            <GifExpertApp />
        )

        const input = screen.getByRole('textbox');

        fireEvent.input(input, { target: { value: inputValue } })

        expect(input.value).toBe(inputValue)
    })

    test('acción de submit del formulario', () => {

        render(
            <GifExpertApp />
        )

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form')

        fireEvent.input(input, { target: { value: inputValue } })

        fireEvent.submit(form)
        expect(input.value).toBe('')
    })

    test('prueba de comparación con el snapshot', () => {

        const { container } = render(
            <GifExpertApp />
        )

        expect(container).toMatchSnapshot();

    })



})