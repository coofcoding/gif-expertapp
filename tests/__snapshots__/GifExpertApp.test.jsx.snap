// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> prueba de comparaci√≥n con el snapshot 1`] = `
<div>
  <div
    class="search-bar"
  >
    <h1>
      Search Gifs
    </h1>
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    class="category-section"
  >
    <h3>
      category: 
      <span
        class="category-style"
      >
        fnaf
      </span>
    </h3>
    <h2
      class="loading"
    >
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
  </div>
</div>
`;
